src/Components/test/E2ETest/Tests/ClientSideHostingTest.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.E2ETesting;
+using Microsoft.AspNetCore.Hosting;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.Tests
+{
+    /// <summary>
+    /// Tests for various MapFallbackToClientSideBlazor overloads. We're just verifying that things render correctly.
+    /// That means that routing and file serving is working for the startup pattern under test.
+    /// </summary>
+    public class ClientSideHostingTest : ServerTestBase<AspNetSiteServerFixture>
+    {
+        public ClientSideHostingTest(
+            BrowserFixture browserFixture,
+            AspNetSiteServerFixture serverFixture,
+            ITestOutputHelper output)
+            : base(browserFixture, serverFixture, output)
+        {
+            serverFixture.ApplicationAssembly = typeof(TestServer.Program).Assembly;
+            serverFixture.BuildWebHostMethod = BuildWebHost;
+        }
+
+        private IWebHost BuildWebHost(string[] args)
+        {
+            return TestServer.Program.BuildWebHost<TestServer.StartupWithMapFallbackToClientSideBlazor>(args);
+        }
+
+        [Fact]
+        public void MapFallbackToClientSideBlazor_FilePath()
+        {
+            Navigate("/filepath", noReload: true);
```

Comment:
I think we should have `noReload: false` in all these cases.

I know that in practice, it probably is going to reload in all cases because the URL is changing, but we don't need to couple the tests to that inference. We do want it to reload in all cases so that the startup logic is tested.

Also I appreciate it's a bit annoying that the condition `noReload` is phrased negatively and it's unclear what the default is.